modules/platforms/dotnet/Apache.Ignite.Core/Impl/Deployment/PeerAssemblyResolver.cs

```diff
@@ -36,16 +37,29 @@ internal sealed class PeerAssemblyResolver : IDisposable
         /** Assembly resolve handler. */
         private readonly ResolveEventHandler _handler;
 
+        /** DataSlot name to be examined during peer loading  */
+        private const string DisablePeerLoadingMark = "DisablePeerLoading";
+
         /// <summary>
         /// Initializes a new instance of the <see cref="PeerAssemblyResolver"/> class.
         /// </summary>
         public PeerAssemblyResolver(IIgniteInternal ignite, Guid originNodeId)
         {
             Debug.Assert(ignite != null);
 
-            _handler = (sender, args) => GetAssembly(ignite, args.Name, originNodeId);
+            _handler = (sender, args) =>
+            {
+                object result = CallContext.LogicalGetData(DisablePeerLoadingMark);
```

Comment:
Probably not a best one towards Net standard, but does reduce amount of code changes for now

Reply:
This does not compile on .NET Core / .NET Standard.
Let's use `ThreadStatic` instead.

Please move the logic to `GetAssembly` method as well to keep the lambda small.

Reply:
There are multiple calls to `GetAssembly` method , the goal was to reduce an amount of auxiliary checks for all cases except the particular one. I think we can extract a helper method for a better readability. Any thoughts?